firmitas/base/difftool.py

```diff
@@ -0,0 +1,118 @@
+"""
+Firmitas
+Copyright (C) 2023-2024 Akashdeep Dhar
+
+This program is free software: you can redistribute it and/or modify it under
+the terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later
+version.
+
+This program is distributed in the hope that it will be useful, but WITHOUT
+ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+details.
+
+You should have received a copy of the GNU General Public License along with
+this program.  If not, see <https://www.gnu.org/licenses/>.
+
+Any Red Hat trademarks that are incorporated in the source code or
+documentation are not subject to the GNU General Public License and may only
+be used or replicated with the express permission of Red Hat, Inc.
+"""
+
+
+from pathlib import Path
+
+import yaml
+from cryptography import x509
+from cryptography.hazmat.backends import default_backend
+
+from firmitas.base.maintool import readcert
+from firmitas.conf import logrdata, standard
+
+
+class Difference:
+    def __init__(self):
+        # Populate a list of certificate files from the device
+        self.certlist_from_device = []
+        certloca = Path(standard.certloca)
+        for item in certloca.iterdir():
+            if not item.is_file() or ".crt" not in item.name:
+                continue
+            self.certlist_from_device.append(item.name)
+
+        # Populate a list of certificate files from the config
+        self.certlist_from_config = []
+        self.certdict = yaml.safe_load(Path(standard.hostloca).read_text())
+        for item in self.certdict:
+            self.certlist_from_config.append(self.certdict[item]["path"])
```

Comment:
Same here, and you can use `.values()` to iterate only over the values of a `dict`:

```suggestion
        self.certdict = yaml.safe_load(Path(standard.hostloca).read_text())
        self.certlist_from_config = [item["path"] for item in self.certdict.values()]
```